src/mmw/js/src/core/layerPicker.js

```diff
@@ -0,0 +1,274 @@
+'use strict';
+
+var models = require('./models'),
+    L = require('leaflet'),
+    $ = require('jquery'),
+    Marionette = require('../../shim/backbone.marionette'),
+    Backbone = require('backbone'),
+    _ = require('underscore'),
+    utils = require('./utils'),
+    layerPickerTmpl = require('./templates/layerPicker.html'),
+    layerPickerGroupTmpl = require('./templates/layerPickerGroup.html'),
+    layerPickerLayerTmpl = require('./templates/layerPickerLayer.html');
+
+/* The individual layers in each layer group */
+var LayerPickerLayerView = Marionette.ItemView.extend({
+    template: layerPickerLayerTmpl,
+
+    ui: {
+        layerHelpIcon: '[data-layer-help-icon]',
+    },
+
+    modelEvents: {
+        'change': 'render'
+    },
+
+    triggers: {
+        'click button': 'select:layer',
+    },
+
+    templateHelpers: function() {
+        return {
+            layerDisplay: this.model.get('display'),
+            layerClass: this.model.get('active') ? 'layerpicker-title active' : 'layerpicker-title',
+            isDisabled: this.model.get('disabled'),
+        };
+    },
+
+    onRender: function() {
+        this.ui.layerHelpIcon.popover({
+            trigger: 'hover',
+            viewport: '.map-container'
+        });
+    },
+});
+
+/* The list of layers in a layer group */
+var LayerPickerLayerListView = Marionette.CollectionView.extend({
+    childView: LayerPickerLayerView,
+    onChildviewSelectLayer: function(childView) {
+        this.toggleLayer(childView.model);
+    },
+    initialize: function(options) {
+        this.leafletMap = options.leafletMap;
+        utils.zoomToggle(this.leafletMap, options.collection.toJSON(),
+            _.bind(this.updateDisabled, this), _.bind(this.clearBgBufferOnLayer, this));
+        utils.perimeterToggle(this.leafletMap, options.collection.toJSON(),
+            _.bind(this.updateDisabled, this), _.bind(this.clearBgBufferOnLayer, this));
+    },
+
+    updateDisabled: function(layer, shouldDisable) {
+        this.collection.findWhere({ display: layer.display })
+            .set('disabled', shouldDisable);
```

Comment:
I'm not seeing this `disabled` property translate to the rendered template for `minZoom` invalidations, I think this feature got dropped.  I can show you quickly what the expected behavior is.